### ISSUE TITLE
IPP Everywhere and Canon GX7000 series: can only print single page

### ISSUE THREAD

**Issue by cyrix2000** [OpenPrinting/cups-filters#490]
Hi,
since the last firmware update of my Canon GX7050 printer which is connected via IPP to my CUPS server (RPi4, Ubuntu 20.04), it only prints a single page. E.g. if I'm sending a two page document, it stops after page 1, or if I have set it to duplex mode, it turns the page but doesn't print anything on the back. Prior to the firmware update it was working fine. connected to a windows PC, it prints without any issues.

**Comment by metabiswadeep** [comment]
Hi @cyrix2000 , could you please share the log file in debug mode of cups. Also please provide more information about the file you printed. If possible please share the file you printed.

**Comment by cyrix2000** [comment]
Hi @metabiswadeep,
Please find attached the file I printed and the error log
[demo.pdf](https://github.com/OpenPrinting/cups-filters/files/10257500/demo.pdf)
[error_log.zip](https://github.com/OpenPrinting/cups-filters/files/10257502/error_log.zip)

It happens for every document if printed in duplex mode: it only prints the first page, then turns the page but doesn't print the second page, just throws out the paper empty.

Thanks for looking into this.

**Comment by metabiswadeep** [comment]
@cyrix2000 does this also happen every time you print documents of more than one page in single page mode as well?

**Comment by cyrix2000** [comment]
If I'm trying to print a multi-page document in single page mode it just stops after page one.

**Comment by tillkamppeter** [comment]
@cyrix2000 What is "single page" mode?

And could you attach the PPD file of your print queue (`/etc/cups/ppd/NAME.ppd`, with `NAME` being the print queue name). Please rename the copy of the file adding `.txt` to the end of its file name, so that it gets accepted as attachment. Do not compress any file you attach.

**Comment by metabiswadeep** [comment]
@tillkamppeter From what I gather from the error log file, after the connection is established, it gets disconnected soon after due to passing of some invalid arguments. The printer then bounces back to some default function which then prints the first page of the document assuming it to be a test page. So what is needed to be done here is to find the argument due to which the connection gets closed before the print job has finished?

**Comment by cyrix2000** [comment]
@metabiswadeep
With "single page mode" I meant "not duplex" lol. Attached the ppd file.
[Canon_GX7050_Duplex.txt](https://github.com/OpenPrinting/cups-filters/files/10318845/Canon_GX7050_Duplex.txt)

**Comment by metabiswadeep** [comment]
@cyrix2000 Could you please add  `AllowExpiredCerts Yes` in `/etc/cups/client.conf` and try printing the document again?

**Comment by cyrix2000** [comment]
@metabiswadeep I tried it (had to generate the client.conf file, my cups install didn't have it, it seems to be depreciated in newer versions (?)), but it made no difference.

**Comment by metabiswadeep** [comment]
@cyrix2000 Please also try deleting any old printer certificate from `/etc/cups/ssl` and reprinting

**Comment by metabiswadeep** [comment]
@cyrix2000 Could you please share the new debug file generated after implantation of the above-mentioned changes.

**Comment by cyrix2000** [comment]
@metabiswadeep here the new debug log
[cups_debug.zip](https://github.com/OpenPrinting/cups-filters/files/10347955/cups_debug.zip)

**Comment by metabiswadeep** [comment]
@cyrix2000 Did you try uninstalling and reinstalling cups?

**Comment by cyrix2000** [comment]
Yes, that was the first thing I tried. At the moment I'm using the version which is part of Raspbian 11 (v2.3.3op2)

**Comment by metabiswadeep** [comment]
@cyrix2000 Have you tried declaring the page numbers to be printed on the command-line? If not, please try using `page-ranges=1-2`

**Comment by metabiswadeep** [comment]
@cyrix2000 Did you try it?

**Comment by cyrix2000** [comment]
@metabiswadeep I'm at the moment away from the printer for the next two weeks. I'll try it when I'm back.

**Comment by tillkamppeter** [comment]
@cyrix2000 Any news?

**Comment by cyrix2000** [comment]
Hi,
sorry I totally missed that. I tried it with -o page-ranges=1-2 and still same issue. It only prints page one, turns the page and outputs it empty again.
